stl/inc/bit

```diff
@@ -125,17 +234,17 @@ _NODISCARD constexpr int countr_one(const _Ty _Val) noexcept {
 
 template <class _Ty, enable_if_t<_Is_standard_unsigned_integer<_Ty>, int> _Enabled = 0>
 _NODISCARD constexpr int popcount(const _Ty _Val) noexcept {
-    if constexpr (sizeof(_Ty) <= sizeof(unsigned int)) {
-        return __builtin_popcount(_Val);
-    } else {
-        return __builtin_popcountll(_Val);
+    if (!_STD is_constant_evaluated()) {
+#if defined(_M_IX86) || defined(_M_X64)
+        return _Checked_x86_x64_popcount(_Val);
+#endif // defined(_M_IX86) || defined(_M_X64)
     }

```

Comment:
Would it make sense to expand the scope of this preprocessor conditional to encompass the whole braced if-statement? Then it would read naturally: "if x86/x64, check if runtime evaluated, call intrinsic version if so". And for ARM/ARM64, it wouldn't expand to an unnecessary call to `is_constant_evaluated` with an empty controlled statement.